edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/edgedeployment/PodExtensions.cs

```diff
@@ -83,19 +87,23 @@ static ReportedModuleStatus ConvertPodStatusToModuleStatus(Option<V1PodStatus> p
                                     else if (c.State.Terminated != null)
                                     {
                                         if (c.State.Terminated.ExitCode != 0)
+                                        {
                                             return new ReportedModuleStatus(ModuleStatus.Failed, $"Module Failed reason: {c.State.Terminated.Reason}");
+                                        }   
                                         else
+                                        {
                                             return new ReportedModuleStatus(ModuleStatus.Stopped, $"Module Stopped reason: {c.State.Terminated.Reason}");
+                                        }            
                                     }
                                     else
                                     {
                                         return new ReportedModuleStatus(ModuleStatus.Backoff, $"Started at {c.State.Running.StartedAt}");
                                     }
-                                }).GetOrElse(() => new ReportedModuleStatus(ModuleStatus.Failed, $"Module Failed with container status Unknown More Info: K8s reason: {status.Reason} with message: {status.Message}"));
+                                }).GetOrElse(() => new ReportedModuleStatus(ModuleStatus.Failed, $"Module Failed with container status Unknown More Info: K8s reason: {status.Conditions.Last().Reason} with message: {status.Conditions.Last().Message}"));
```

Comment:
Are you somehow guaranteeing order in your Enumerable here such that you can depend on .Last? Why not .First or if this Enumerable has only one item, Single? Also, IIRC, Last/Single/First will throw if the Enumerable is empty... is there a chance that it is empty? If so, I sugest you use the OrDefault versions of these functions.

Reply:
Comment applies to all uses of Last.